### ISSUE TITLE
how to print images on printer EPSON TMU 220D?

### ISSUE THREAD

**Issue by Joos94** [mike42/escpos-php#98]
Hi Mike,
I'm using POS printer EPSON TMU220D, I could not print an image on the printer.
there is a function that allows the printing of the image?
or it is not possible in this type of printer?

**Comment by mike42** [comment]
Graphics are supported on the TM U220. The trick is to use single-density graphics only, and a much older print command.

Quoting #35 

````php
$tux = new EscposImage("resources/tux.png")
$printer -> bitImageColumnFormat($tux, Escpos::IMG_DOUBLE_WIDTH | Escpos::IMG_DOUBLE_HEIGHT);
````

Tips-
- The `bitImageColumnFormat` method is a recent addition, make sure you are using the current `master` branch (or `dev-master` version in composer-land).
- My U220 struggles to keep up with long images (prints some garbage if the computer sends lines of image too quickly), so I suggest either keeping it simple, or being prepared to inject some `sleep()` statements in to `bitImageColumnFormat` if you are lucky enough to be using a `FilePrintConnector`.

**Comment by theslowblitz** [comment]
Hi Mike, I'm using a TM-U220B to print tux.png. I get gibberish when printing with graphics(), bitImage(), and bitImageColumnFormat(). For bitImageColumnFormat, my parameters were bitImageColumnFormat($tux) instead of bitImageColumnFormat($tux, Escpos::IMG_DOUBLE_WIDTH | Escpos::IMG_DOUBLE_HEIGHT), since my program couldn't find the Escpos class? Text prints great, so I'm not sure what I'm doing wrong.

**Comment by mike42** [comment]
Version 1.2 (released Apr 10 2016) included commit 560291e9dabb0c65bc2dc39f115bebf02606a8f8, which updated references to `Escpos`, which is now called `Printer`. I've updated the comment above to reflect this.

[This example](https://github.com/mike42/escpos-php/blob/35d9a2c7fbf9c1d00b6665bf3f362e4a8fd55eec/example/bit-image.php) shows some of the plumbing which will apply to you.

Please open a fresh issue if you run into any other troubles, so that your questions aren't lost.
